python/cudf/cudf/core/dataframe.py

```diff
@@ -1924,9 +1926,10 @@ def as_gpu_matrix(self, columns=None, order="F"):
         ncol = len(cols)
         nrow = len(self)
         if ncol < 1:
-            raise ValueError("require at least 1 column")
-        if nrow < 1:
-            raise ValueError("require at least 1 row")
+            # This is the case for empty dataframe - construct empty cupy array
+            matrix = cupy.empty(shape=(0, 0), order=order)
```

Comment:
But in case of an empty dataframe, I don't think we'll have any dtypes to choose from right? or do we default to something like `float64` in that case?

Reply:
Sorry, if we have no columns then yes we don't have anything to go on. What does Pandas do in this situation?

Reply:
From a quick test looks like they default to `float64` so lets do that 😄